seed-bank-dynamics/scripts/download-data.py

```diff
@@ -1,32 +1,35 @@
-"""Downloads a data file from a given URL.
-
-This is the most basic of the download data scripts. It requires that the url for
-the data is provided as an argument, and it assumes that a data-raw folder
-has been created containing a .gitignore file.
-"""
-
 from pathlib import Path
-
 import requests
 
 resource_dir = Path(__file__).resolve().parent.parent
-
 folder_path = resource_dir / "data-raw"
 
-url3a = "https://zenodo.org/records/4965431/files/Expe3Abundance.csv?download=1"
-
-raw_data3a = requests.get(url3a, allow_redirects=True)
-
-file_path = folder_path / "data3a.csv"
-
-with open(file_path, "wb") as file:
-    file.write(raw_data3a.content)
-
-url3b = "https://zenodo.org/records/4965431/files/Expe3Richness.csv?download=1"
-
-raw_data3b = requests.get(url3b, allow_redirects=True)
-
-file_path = folder_path / "data3b.csv"
-
-with open(file_path, "wb") as file:
-    file.write(raw_data3b.content)
+def download_data(url, name):
+    try:      
+        # Get the data from the URL
+        raw_data = requests.get(url, allow_redirects=True)
+        raw_data.raise_for_status()  # Raise an exception for HTTP errors
+
+        file_path = folder_path / name
+        with open(file_path, "wb") as file:
+            file.write(raw_data.content)
+
+    except requests.exceptions.RequestException as e:
+        print(f"Error downloading the data: {e}")
+    except Exception as e:
+        print(f"An unexpected error occurred: {e}")
```

Comment:
Not sure this level of engineering for this type of function and its uses is necessary, but it's good practice :tada: